ops/testnet.py

```diff
@@ -0,0 +1,153 @@
+from os import path, mkdir
+from subprocess import run
+from socket import socket
+from tempfile import gettempdir
+from shutil import rmtree
+
+from config import CONFIG
+
+def generateClientTOML(tmpDir, authIP, startingPortNumber, authorityPublicKey, decoyTraffic="true"):
+    with open(path.join(tmpDir, "client.toml"), 'w+') as f:
+        f.write(
+            """
+            [Logging]
+              Disable = false
+              Level = "DEBUG"
+              File = ""
+
+            [UpstreamProxy]
+              Type = "none"
+
+            [Debug]
+              DisableDecoyTraffic = {}
+              CaseSensitiveUserIdentifiers = false
+              PollingInterval = 1
+
+            [NonvotingAuthority]
+                Address = "{}:{}"
+                PublicKey = "{}"
+            """.format(decoyTraffic, authIP, startingPortNumber, authorityPublicKey)
+        )
+
+# get ip address
+s = socket()
+s.connect(("1.1.1.1", 80))
+ip = s.getsockname()[0]
+print("Accesible IP address: ", ip)
+s.close()
+
+testnetConfDir = path.join(gettempdir(), "meson-testnet")
+try: rmtree(testnetConfDir)
+except: pass
+mkdir(testnetConfDir)
+
+run([
+    "genconfig",
+    "-o",
+    testnetConfDir,
+    "-n",
+    str(CONFIG["TEST"]["NODES"]),
+    "-a",
+    ip,
+    "-p",
+    str(CONFIG["TEST"]["PROVIDERS"]),
+], check=True)
+
+# open some files to read their contents
+with open(path.join(testnetConfDir, "nonvoting", "identity.public.pem"), 'r') as f:
+    authorityPublicKey = f.read().split("\n")[1] # line index 1
+
+with open(path.join(testnetConfDir, "nonvoting", "authority.toml"), 'r') as f:
+    for line in f:
+        if "Addresses = [" in line:
+            startingPortNumber = line.split('"')[1].split(":")[1]
+            break
+
+with open(path.join(testnetConfDir, "provider-0", "katzenpost.toml"), 'r') as f:
+    for line in f:
+        if "UserRegistrationHTTPAddresses" in line:
+            startingUserRegistrationPort = line.split('"')[1].split(":")[1]
+            break
+
+generateClientTOML(testnetConfDir, ip, startingPortNumber, authorityPublicKey)
+
+composeYMLFile = """
```

Comment:
Maybe create a utils file where you have all these cookie-cutter file generation utilities and other stuff across files.